core/worker.go

```diff
@@ -763,8 +763,16 @@ func (w *worker) prepareWork(genParams *generateParams, block *types.Block) (*en
 			} else {
 				header.SetParentDeltaS(w.engine.DeltaLogS(parent.Header()), nodeCtx)
 			}
+			// update only on prime block
+			if order == common.PRIME_CTX && nodeCtx == common.ZONE_CTX {
+				header.SetTerminusHash(block.Header().Hash())
+			} else if nodeCtx == common.ZONE_CTX {
```

Comment:
No need to take care of region case?

Reply:
We are only using these to create variance that is consistant within a slice for the duration of a prime, so they are only updated in the zone on a prime.  So yes, no region blocks.